mypyc/irbuild/specialize.py

```diff
@@ -256,3 +258,12 @@ def translate_isinstance(builder: IRBuilder, expr: CallExpr, callee: RefExpr) ->
         if irs is not None:
             return builder.builder.isinstance_helper(builder.accept(expr.args[0]), irs, expr.line)
     return None
+
+
+@specialize_function('builtins.min')
+def faster_min(builder: IRBuilder, expr: CallExpr, callee: RefExpr):
```

Comment:
You only need to use `@specialize_function` if you inline the body of `min` in the generated IR (see below). Otherwise using `c_function_op` should be sufficient.